File: 34xxx/34766. 쿠의 용돈 인상 계획/쿠의 용돈 인상 계획.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.StringTokenizer;

public class Main {
	static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
	static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));	
	public static void main(String[] args) throws IOException {
		int t = Integer.valueOf(br.readLine());
		
		while(t-->0) {
			StringTokenizer st = new StringTokenizer(br.readLine());
			int d = Integer.valueOf(st.nextToken());
			int k = Integer.valueOf(st.nextToken());
			String string = br.readLine();
			
			int idx=0;
			for(int i=0;i<d;i++) {
				if(string.charAt(i)>'1') idx=i;
			}
			for(int i=0;i<idx;i++) bw.write(string.charAt(i));
			
			int cur = string.charAt(idx)-'0';
			int remain=0;
			while(k-->0) {
				cur*=9;
				if(cur==81) {
					remain++;
					cur=8;
				} else if(cur>=10) {
					bw.write(Integer.toString(cur/10));
					cur%=10;
				}
			}
			bw.write(Integer.toString(cur));
			while(remain-->0) bw.write('1');
			for(int i=idx+1;i<d;i++) bw.write(string.charAt(i));
			bw.write("\n");
		}
		
        bw.close();
    } // end of main
} // end of Main

File: 34xxx/34766. 쿠의 용돈 인상 계획/쿠의 용돈 인상 계획.cc
#include<bits/stdc++.h>
using namespace std;

int main() {
    ios::sync_with_stdio(0); cin.tie(0);
    int t; cin >> t;
    while(t--) {
        int d, k; string s; cin >> d >> k >> s;
        int idx=0;
        for(int i=1;i<d;i++) {
            if(s[i]>'1') idx=i;
        }

        for(int i=0;i<idx;i++) cout << s[i];
        int cur = s[idx]-'0';
        int remain=0; // 81
        while(k--) {
            if(cur==1) cur=9;
            else {
                cur*=9;
                if(cur==81) {
                    remain++;
                    cur=8;
                } else {
                    cout << cur/10;
                    cur%=10;
                }
            }
        }
        cout << cur;
        while(remain--) cout << 1;
        for(int i=idx+1;i<d;i++) cout << s[i];
        cout << '\n';
    }
}